### ISSUE TITLE
Register component is broken

### ISSUE THREAD

**Issue by stevepiercy** [plone/volto#6233]
@Tishasoumya-02 The refactor looks good here, but the actual component seems like it was broken before refactoring.

It is unable to actually succeed in a form submission due to the backend call return an error as the form doesn't send a `password` value or the `sendPasswordReset` option. If you could write a test that should pass (i.e. the form sends successfully and the toast appears) but actually fails that would be great and we can go from there on getting this merged.

cc @sneridagh @nileshgulia1 For visibility

## Returned error

```
{
  "error": {
    "errors": [
      {
        "field": "username", 
        "message": "Property 'User Name' is required."
      }, 
      {
        "field": "sendPasswordReset", 
        "message": "You have to either send a password or sendPasswordReset."
      }
    ], 
    "message": "Error in fields. Property 'User Name' is required. You have to either send a password or sendPasswordReset.", 
    "type": "WrongParameterError"
  }
}
```

_Originally posted by @JeffersonBledsoe in https://github.com/plone/volto/issues/4861#issuecomment-1627396334_

**Comment by MahtabShah** [comment]
Hey can i contribute ?

**Comment by stevepiercy** [comment]
@MahtabShah please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by bint-Eve** [comment]
can i work on this ?

**Comment by stevepiercy** [comment]
@bint-Eve no. Please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do, Item 2](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).
